edb/edgeql/declarative.py

```diff
@@ -1078,6 +1089,14 @@ def _get_pointer_deps(
     if pointer in ctx.objects:
         result.add(pointer)
 
+    # HACK: Add all pointers that have this pointer (link, actually)
```

Comment:
I'm a bit confused here; why doesn't the property dependency get picked up normally by tracing?

Reply:
To pick up a link property we need the tracer to be aware when the context of the expression is a path that denotes a link, currently the tracer only concerns itself with the link target though.

Ideally we'd want the tracer to be better, but as I said in my overall comment this might be a good reason to change the entire approach rather than try to teach the existing tracer about the link properties. The existing tracer is trying to get things approximately right with less deep analysis.

Reply:
Oh, roger, got it.